recipe/meta.yaml

```diff
@@ -1,35 +1,167 @@
-{% set name = "openfst" %}
 {% set version = "1.8.2" %}
 
 package:
-  name: {{ name|lower }}
+  name: openfst
   version: {{ version }}
 
 source:
-  url: http://www.openfst.org/twiki/pub/FST/FstDownload/{{ name }}-{{ version }}.tar.gz
+  url: http://www.openfst.org/twiki/pub/FST/FstDownload/openfst-{{ version }}.tar.gz
   sha256: de987bf3624721c5d5ba321af95751898e4f4bb41c8a36e2d64f0627656d8b42
   patches:
-    - osx_arm64_configure.patch  # [build_platform != target_platform]
+    - patches/0001-Remove-float-equality-check-in-cross-compiling.patch  # [build_platform != target_platform]
+    - patches/0002-add-missing-header.patch
+    - patches/0003-add-no-undefined-to-some-LDFLAGS.patch                # [win]
+    - patches/0004-Cross-platform-CMake-build.patch                      # [not linux]
+    - patches/0005-Update-installation-of-fsts-headers.patch             # [linux]
 
 build:
-  number: 1
-  skip: true  # [win]
+  number: 2
+  # ensures that packages using openfst as a host dependency get the right runtime dependence as well, see
+  # https://conda-forge.org/docs/maintainer/pinning_deps.html#specifying-run-exports
+  run_exports:
+  # seems to be changing even with patch releases
+  # https://abi-laboratory.pro/tracker/timeline/openfst/
+    - {{ pin_subpackage('openfst', max_pin='x.x.x') }}
 
 requirements:
   build:
     - {{ compiler('cxx') }}
-    - make
-    - gnuconfig   # [unix]
+    - make         # [linux]
+    - gnuconfig    # [linux]
+    - cmake        # [not linux]
+    - ninja        # [not linux]
+    - m2-patch     # [win]
+  host:
+    - dlfcn-win32  # [win]
+  run:
+    - ucrt         # [win]
+
+{% set fst_binaries = [
+    "farcompilestrings", "farconvert", "farcreate", "farencode", "farequal", "farextract",
+    "farinfo", "farisomorphic", "farprintstrings", "fstarcsort", "fstclosure", "fstcompile",
+    "fstcompose", "fstcompress", "fstconcat", "fstconnect", "fstconvert", "fstdeterminize",
+    "fstdifference", "fstdisambiguate", "fstdraw", "fstencode", "fstepsnormalize", "fstequal",
+    "fstequivalent", "fstinfo", "fstintersect", "fstinvert", "fstisomorphic", "fstlinear",
+    "fstloglinearapply", "fstmap", "fstminimize", "fstprint", "fstproject", "fstprune", "fstpush",
+    "fstrandgen", "fstrelabel", "fstreplace", "fstreverse", "fstreweight", "fstrmepsilon",
+    "fstshortestdistance", "fstshortestpath", "fstsymbols", "fstsynchronize",
+    "fsttopsort", "fstunion", "mpdtcompose", "mpdtexpand", "mpdtinfo", "mpdtreverse",
+    "pdtcompose", "pdtexpand", "pdtinfo", "pdtreplace", "pdtreverse", "pdtshortestpath"
+] %}
+{% set fst_libraries = [
+    "", "compact", "compressscript", "const", "far", "farscript",
+    "linearscript", "lookahead", "mpdtscript", "ngram", "pdtscript",
+    "script", "special"
+] %}
+# This list should be substantially longer; this is just a stand-in
+# selection; relevant headers can be added as needed/desired.
+{% set fst_headers = [
```

Comment:
@mmcauliffe 
It sounds like this comment is now out-of-date. Did you add all headers or just a large subset?

Reply:
Ah yes, all headers should be included now, so we can get rid of that comment